cmd/fleet/vuln_process.go

```diff
@@ -0,0 +1,162 @@
+			instanceID, err := server.GenerateRandomText(64)
+			if err != nil {
+				initFatal(errors.New("error generating random instance identifier"), "")
+			}
+			// using the same lock name as the cron scheduled version of vuln processing, that way if we fail to obtain the lock
+			// it's most likely due to vulnerabilities.disable_schedule=false but still trying to run external vuln processing command
+			lock, err := ds.Lock(ctx, string(fleet.CronVulnerabilities), instanceID, lockDuration)
+			if err != nil {
+				initFatal(err, "failed to obtain vuln processing lock")
+			}
+			if !lock {
+				initFatal(errors.New("vulnerabilities processing locked"),
+					"failed to obtain vuln processing lock, something else still has lock ownership")
+			}
+			defer func() {
+				err = ds.Unlock(ctx, "cron_vulnerabilities", "vuln_processing_command")
+				if err != nil {
+					initFatal(err, "failed to release vuln processing lock")
+				}
+				cancel()
+			}()
+
+			logger := initLogger(cfg)
+			logger = kitlog.With(logger, fleet.CronVulnerabilities)
+
+			appConfig, err := ds.AppConfig(ctx)
+			if err != nil {
+				initFatal(err, "error fetching app config during vulnerability processing")
+			}
+			vulnConfig := cfg.Vulnerabilities
+			vulnPath := configureVulnPath(vulnConfig, appConfig, logger)
+			// this really shouldn't ever be empty string since it's defaulted, but could be due to some misconfiguration
+			// we'll throw an error here since the entire point of this command is to process vulnerabilities
+			if vulnPath == "" {
+				initFatal(errors.New("vuln path empty, check environment variables or app config yml"), "error during vulnerability processing")
+			}
+			level.Info(logger).Log("msg", "scanning vulnerabilities")
+			err = scanVulnerabilities(ctx, ds, logger, &vulnConfig, appConfig, vulnPath)
+			if err != nil {
+				// errors during vuln processing should bubble up, so you know the job is failing without having to scour logs, e.g. non-zero exit code
+				initFatal(fmt.Errorf("scanning vulnerabilities: %w", err), "error during vulnerability processing")
+			}
+
+			err = ds.SyncHostsSoftware(ctx, time.Now())
+			if err != nil {
+				// though vulnerability processing succeeded, we'll still fatally error here to indicate there was a problem
+				initFatal(err, "failed to sync host software")
+			}
+
+			return
+		},
+	}
+	vulnProcessingCmd.PersistentFlags().BoolVar(&dev, "dev", false, "Enable developer options")
+	vulnProcessingCmd.PersistentFlags().BoolVar(&devLicense, "dev_license", false, "Enable development license")
+	vulnProcessingCmd.PersistentFlags().BoolVar(&devExpiredLicense, "dev_expired_license", false, "Enable expired development license")
+	vulnProcessingCmd.PersistentFlags().DurationVar(
+		&lockDuration,
+		"lock_duration",
+		time.Second*60*60,
+		"the duration (https://pkg.go.dev/time#ParseDuration) the lock should be obtained, ideally this duration is less than the interval in which the job runs (defaults to 60m)")
```

Comment:
One last thing, let's now document here that the job will exit if it doesn't finish by `lock_duration` time.
And that this is the default because we've seen vulnerability processing take usually less than one hour.